src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsalertsapi/resource/MigrateAlertsController.kt

```diff
@@ -0,0 +1,75 @@
+package uk.gov.justice.digital.hmpps.hmppsalertsapi.resource
+
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.Parameter
+import io.swagger.v3.oas.annotations.media.Content
+import io.swagger.v3.oas.annotations.media.Schema
+import io.swagger.v3.oas.annotations.responses.ApiResponse
+import io.swagger.v3.oas.annotations.responses.ApiResponses
+import jakarta.servlet.http.HttpServletRequest
+import jakarta.validation.Valid
+import org.springframework.http.HttpStatus
+import org.springframework.http.MediaType
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.ResponseStatus
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.hmppsalertsapi.model.Alert
+import uk.gov.justice.digital.hmpps.hmppsalertsapi.model.request.MigrateAlertRequest
+import uk.gov.justice.digital.hmpps.hmppsalertsapi.service.MigrateAlertService
+import uk.gov.justice.hmpps.kotlin.common.ErrorResponse
+
+@RestController
+@RequestMapping("/migrate/alerts", produces = [MediaType.APPLICATION_JSON_VALUE])
+class MigrateAlertsController(
+  private val migrateAlertService: MigrateAlertService,
+) {
+
+  @ResponseStatus(HttpStatus.CREATED)
+  @PostMapping
+  @Operation(
+    summary = "Create an alert",
+  )
+  @ApiResponses(
+    value = [
+      ApiResponse(
+        responseCode = "201",
+        description = "Alert created successfully",
+        content = [Content(schema = Schema(implementation = Alert::class))],
+      ),
+      ApiResponse(
+        responseCode = "400",
+        description = "Bad request",
+        content = [Content(schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorised, requires a valid Oauth2 token",
+        content = [Content(schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Forbidden, requires an appropriate role",
+        content = [Content(schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "409",
+        description = "Conflict, the person already has an active alert using the supplied alert code",
+        content = [Content(schema = Schema(implementation = ErrorResponse::class))],
+      ),
+    ],
+  )
+  @PreAuthorize("hasAnyRole('$ROLE_ALERTS_WRITER', '$ROLE_ALERTS_ADMIN', '$UPDATE_ALERT', '$ROLE_NOMIS_ALERTS')")
```

Comment:
The only roles that should be able to call this endpoint are ROLE_ALERTS_ADMIN and ROLE_NOMIS_ALERTS

Reply:
Yes - I was going to check that with you.